cfy_manager/components/restservice/restservice.py

```diff
@@ -403,6 +398,92 @@ def _configure_db_proxy(self):
         service.restart('haproxy', append_prefix=False)
         self._wait_for_haproxy_startup()
 
+    def handle_haproxy_certificate(self):
+        certificates.use_supplied_certificates(
+            logger=self.logger,
+            ca_destination='/etc/haproxy/ca.crt',
+            owner='haproxy',
+            group='haproxy',
+            component_name='postgresql_client'
+        )
+
+    @staticmethod
+    def handle_ldap_certificate():
+        certificates.use_supplied_certificates(
+            logger=logger,
+            ca_destination=LDAP_CA_CERT_PATH,
+            component_name=RESTSERVICE,
+            sub_component='ldap',
+            just_ca_cert=True
+        )
+
+    def replace_certificates(self):
+        if common.manager_using_db_cluster():
+            self._replace_haproxy_cert()
+        self._replace_ldap_cert()
+        self._replace_ca_certs_on_db()
+        service.restart(RESTSERVICE)
+        self._verify_restservice_alive()
+
+    def validate_new_certs(self):
+        # All other certs are validated in other components
```

Comment:
Is there any situation in which we can end up hitting validate+replace on this component without other components having checked the certs? I'm guessing not because the ones it cares about are on nginx?

Reply:
Correct, the other certificates are: 
1. Postgresql CA cert - validated in the postgresql_client component.
2. Internal CA cert - validated in Nginx component.
3. RabbitMQ CA cert - validated in Manager component.